src/components/AppSettingsDialog/AppSettingsDialog.vue

```diff
@@ -0,0 +1,244 @@
+<!--
+ - @copyright Copyright (c) 2020 Marco Ambrosini <marcoambrosini@pm.me>
+ -
+ - @author Marco Ambrosini <marcoambrosini@pm.me>
+ -
+ - @license GNU AGPL version 3 or any later version
+ -
+ - This program is free software: you can redistribute it and/or modify
+ - it under the terms of the GNU Affero General Public License as
+ - published by the Free Software Foundation, either version 3 of the
+ - License, or (at your option) any later version.
+ -
+ - This program is distributed in the hope that it will be useful,
+ - but WITHOUT ANY WARRANTY; without even the implied warranty of
+ - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ - GNU Affero General Public License for more details.
+ -
+ - You should have received a copy of the GNU Affero General Public License
+ - along with this program. If not, see <http://www.gnu.org/licenses/>.
+ -
+ -->
+
+<script>
+import Modal from '../Modal'
+import { emit, subscribe, unsubscribe } from '@nextcloud/event-bus'
+import isMobile from '../../mixins/isMobile'
+
+export default {
+
+	name: 'AppSettingsDialog',
+
+	components: {
+		Modal,
+	},
+
+	mixins: [isMobile],
+
+	props: {
+		hideNavigation: {
+			type: Boolean,
+			default: false,
+		},
+	},
+
+	data() {
+		return {
+			open: true,
+			selectedSection: '',
+		}
+	},
+
+	computed: {
+
+		content() {
+			return document.querySelector('.app-settings__content')
+		},
+
+		hasNavigation() {
+			if (this.isMobile || this.hideNavigation) {
+				return false
+			} else {
+				return true
+			}
+		},
+	},
+
+	mounted() {
+		subscribe('toggle-settings', this.toggleSettings)
+		// Emit an event with the initial state of the settings
+		emit('settings-toggled', {
+			open: this.open,
+		})
+
+		// Select first settings section
+		this.selectedSection = this.$slots.default[0].componentOptions.propsData.sectionTitle
+	},
+
+	methods: {
+
+		/**
+		 * Toggle the settings dialog
+		 *
+		 * @param {Boolean} [bool] set the state instead of inverting the current one
+		 */
+		toggleSettings(bool) {
+			this.open = bool
+			const bodyStyles = getComputedStyle(document.body)
+			const animationLength = parseInt(bodyStyles.getPropertyValue('--animation-quick')) || 100
+			setTimeout(() => {
+				emit('settings-toggled', {
+					open: this.open,
+				})
+				// We wait for 1.5 times the animation length to give the animation time to really finish.
+			}, 1.5 * animationLength)
```

Comment:
THis probably should be handled by the Modal component.
So you can listen like @doneOpen or something

Reply:
It's the same as we did for the navigation. Currently the modal dosn't emit any event when the animation ends. What do you suggest?